### ISSUE TITLE
Compiler hangs during execution

### ISSUE THREAD

**Issue by jjl9807** [rust-lang/rust#139197]
I tried this code:

```rust
trait Trait {}

struct W<T>(T);

impl<T, U> Trait for W<(<&Self as IntoIterator>, W<U>)>
where
    W<T>: Trait,
    W<U>: Trait,
{
}

fn impls<T: Trait>() {}

fn main() {
    impls::<W<_>>();
}
```

I expected to see this happen: 

The compilation process should complete successfully, and any errors in the code should be reported with diagnostic messages.

Instead, this happened: 

The compiler hangs indefinitely and appears to never terminate.

### Meta

`rustc --version --verbose`:
```
rustc 1.85.1 (4eb161250 2025-03-15)
binary: rustc
commit-hash: 4eb161250e340c8f48f66e2b929ef4a5bed7c181
commit-date: 2025-03-15
host: x86_64-pc-windows-msvc
release: 1.85.1
LLVM version: 19.1.7
```

<details><summary>Backtrace</summary>
<p>

```
error: expected `::`, found `,`
 --> .\temp.rs:7:48
  |
7 | impl<T, U> Trait for W<(<&Self as IntoIterator>, W<U>)>
  |                                                ^ expected `::`

error: expected `::`, found `,`
 --> .\temp.rs:7:48
  |
7 | impl<T, U> Trait for W<(<&Self as IntoIterator>, W<U>)>
  |                                                ^ expected `::`
  |
help: expressions must be enclosed in braces to be used as const generic arguments
  |
7 | impl<T, U> Trait for W<{ (<&Self as IntoIterator>, W<U>) }>
  |                        +                                 +

error[E0747]: constant provided when a type was expected
 --> .\temp.rs:7:24
  |
7 | impl<T, U> Trait for W<(<&Self as IntoIterator>, W<U>)>
  |                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

```

</p>
</details>

**Comment by moxian** [comment]
<details>
<summary>bisection</summary>

```
Regression in nightly-2024-06-19:
converted 2024-06-18 to 59e2c01c2217a01546222e4d9ff4e6695ee8a1db
converted 2024-06-19 to 8fcd4dd08e2ba3e922d917d819ba0be066bdb005
get_commits_between returning commits, len: 10
* #125500 (Migrate link-arg, link-dedup and issue-26092 run-make tests to rmake format) - Oneirical:bundle-them-up-why-not, r=jieyouxu
* #126279 (Migrate inaccessible-temp-dir, output-with-hyphens and issue-10971-temps-dir run-make tests to rmake) - Oneirical:you-can-run-make-but-cannot-hide, r=jieyouxu
* #126490 (Migrate extern-flag-fun, incremental-debugger-visualiser and incremental-session-fail run-make tests to rmake.rs) - Oneirical:testicide, r=jieyouxu
* #126591 ([perf] More span update benchmarking)  - petrochenkov:upctxt5, r=nnethercote
* #126330 (Return opaque type from PanicInfo::message()) - m-ou-se:panic-message-type, r=Amanieu
* #126049 (Rework feature(precise_capturing) to represent use<...> as a syntactical bound) - compiler-errors:rework-use, r=oli-obk
* #126437 (Migrate issue-64153, invalid-staticlib and no-builtins-lto run-make tests to rmake)  - Oneirical:test-we-forget, r=jieyouxu
* #126630 (Rollup of 5 pull requests)  - GuillaumeGomez:rollup-hlwbpa2, r=GuillaumeGomez
* #126623 (Replace all &DiagCtxt with a DiagCtxtHandle<'_> wrapper type) - oli-obk:do_not_count_errors, r=davidtwco
* #126614 (Uplift next trait solver to rustc_next_trait_solver) - compiler-errors:uplift-next-trait-solver, r=lcnr
```

</details>

Backtraces from the live process:

<details>
<summary> backtrace1</summary>

```
rustc_driver-ebda0089c9512452.dll!RINvMs6_NtCs9VVGc76UbHk_9hashbrown3rawINtB6_8RawTableTINtNtNtCsacCxZy8tNDM_12rustc_middle2ty7context13InternedInSetINtNtCs1vmCzCB8wM7_13rustc_type_ir7ty_info18WithCachedTypeInfoINtNtB1U_7ty_kind6TyKindNtBT_6TyCtxtEEEuEE14
rustc_driver-ebda0089c9512452.dll!RNvMs4_NtNtCsacCxZy8tNDM_12rustc_middle2ty7contextNtB5_13CtxtInterners9intern_ty+0x1f02
rustc_driver-ebda0089c9512452.dll!RINvMs6_NtCs9VVGc76UbHk_9hashbrown3rawINtB6_8RawTableTTNtNtCsacCxZy8tNDM_12rustc_middle2ty2TyNtCs1vmCzCB8wM7_13rustc_type_ir8VariancebEBR_EE14reserve_rehashNCINvNtB8_3map11make_hasherBQ_BR_NtCszUfY7mhKel_10rustc_hash13FxB
rustc_driver-ebda0089c9512452.dll!RNvMs5_NtCskymifpGYJ4S_11rustc_infer5inferNtB5_9InferCtxt19fresh_args_for_item+0x1b2
rustc_driver-ebda0089c9512452.dll!RNvXs_NtNtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits5query9normalizeNtB4_15QueryNormalizerINtNtCs1vmCzCB8wM7_13rustc_type_ir4fold18FallibleTypeFolderNtNtNtCsacCxZy8tNDM_12rustc_middle2ty7context6TyCtxtE11try_fold_ty+0
rustc_driver-ebda0089c9512452.dll!RNvXs_NtNtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits5query9normalizeNtB4_15QueryNormalizerINtNtCs1vmCzCB8wM7_13rustc_type_ir4fold18FallibleTypeFolderNtNtNtCsacCxZy8tNDM_12rustc_middle2ty7context6TyCtxtE11try_fold_ty+0
rustc_driver-ebda0089c9512452.dll!RNvXs1_NtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits7fulfillNtB5_16FulfillProcessorNtNtCsfk1N0zTQ1Ep_21rustc_data_structures17obligation_forest19ObligationProcessor18process_obligation+0x1e29
rustc_driver-ebda0089c9512452.dll!RINvMs0_NtCsfk1N0zTQ1Ep_21rustc_data_structures17obligation_forestINtB6_16ObligationForestNtNtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits7fulfill26PendingPredicateObligationE19process_obligationsNtB1s_16FulfillProcesso
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x36665
rustc_driver-ebda0089c9512452.dll!RNvXNvMNtCsjEcwfPPTRNV_16rustc_hir_typeck7closureNtNtB7_7fn_ctxt6FnCtxt40deduce_closure_signature_from_predicatesNtB2_10MentionsTyINtNtCs1vmCzCB8wM7_13rustc_type_ir5visit11TypeVisitorNtNtNtCsacCxZy8tNDM_12rustc_middle2ty7
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x12cb1
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x27ec
rustc_driver-ebda0089c9512452.dll!RNvNtCshEjB07rFK3d_16rustc_query_impl8plumbing25query_key_hash_verify_all+0x155db
rustc_driver-ebda0089c9512452.dll!RINvNtNtCs8Ehw5TsioVt_18rustc_query_system5query8plumbing17try_execute_queryINtCshEjB07rFK3d_16rustc_query_impl13DynamicConfigINtNtCsfk1N0zTQ1Ep_21rustc_data_structures9vec_cache8VecCacheNtNtCs8yPNJtyUAIM_10rustc_span6def
rustc_driver-ebda0089c9512452.dll!RNvNtCshEjB07rFK3d_16rustc_query_impl8plumbing25query_key_hash_verify_all+0x1de43
rustc_driver-ebda0089c9512452.dll!RINvMs6_NtCs9VVGc76UbHk_9hashbrown3rawINtB6_8RawTableTRjuEE14reserve_rehashNCINvNtB8_3map11make_hasherBQ_uNtCszUfY7mhKel_10rustc_hash13FxBuildHasherE0ECs2JjSXZNO9KT_18rustc_hir_analysis+0x14f24
rustc_driver-ebda0089c9512452.dll!RNvCs2JjSXZNO9KT_18rustc_hir_analysis11check_crate+0xfc1
rustc_driver-ebda0089c9512452.dll!RNvNtCsbLMVCc8CHyP_15rustc_interface6passes25resolver_for_lowering_raw+0x65b1
rustc_driver-ebda0089c9512452.dll!RNvNtCsbLMVCc8CHyP_15rustc_interface6passes8analysis+0x4e
rustc_driver-ebda0089c9512452.dll!RNvMsj_NtCskXJAKvAmGc7_5alloc4syncINtB5_3ArcNtNtCsdQacyuVW3xU_13rustc_session6cstore11CrateSourceE9drop_slowCshEjB07rFK3d_16rustc_query_impl+0x6cf7a
rustc_driver-ebda0089c9512452.dll!RINvNtNtCs8Ehw5TsioVt_18rustc_query_system5query8plumbing17try_execute_queryINtCshEjB07rFK3d_16rustc_query_impl13DynamicConfigINtNtB4_6caches11SingleCacheINtNtNtCsacCxZy8tNDM_12rustc_middle5query5erase6ErasedAhj0_EEKb0_KB
rustc_driver-ebda0089c9512452.dll!RNvCshEjB07rFK3d_16rustc_query_impl12query_system+0x5377
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCNCINvMNtB6_6threadNtB1h_7Builder16spawn_unchecked_NCINvCs35sVv18JLtM_5ctrlc17set_handler_innerNCNvCs13KITJ2Zc2G_17rustc_driver_impl21install_ctrlc
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCINvNtCsbLMVCc8CHyP_15rustc_interface4util26run_in_thread_with_globalsNCINvB1e_31run_in_thread_pool_with_globalsNCINvNtB1g_9interface12run_compiler
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCNCINvMNtB6_6threadNtB1h_7Builder16spawn_unchecked_NCINvCs35sVv18JLtM_5ctrlc17set_handler_innerNCNvCs13KITJ2Zc2G_17rustc_driver_impl21install_ctrlc
rustc_driver-ebda0089c9512452.dll!ZN3std3sys3pal7windows6thread6Thread3new17h0a1b245cf1c341adE+0x1dd
KERNEL32.DLL!BaseThreadInitThunk+0x14
ntdll.dll!RtlUserThreadStart+0x21
```

</details>

<details>
<summary>backtrace 2</summary>

```
rustc_driver-ebda0089c9512452.dll!RNvMs5_NtCskymifpGYJ4S_11rustc_infer5inferNtB5_9InferCtxt15shallow_resolve+0x1
rustc_driver-ebda0089c9512452.dll!RNvXs1_NtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits7fulfillNtB5_16FulfillProcessorNtNtCsfk1N0zTQ1Ep_21rustc_data_structures17obligation_forest19ObligationProcessor18process_obligation+0x1248
rustc_driver-ebda0089c9512452.dll!RINvMs0_NtCsfk1N0zTQ1Ep_21rustc_data_structures17obligation_forestINtB6_16ObligationForestNtNtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits7fulfill26PendingPredicateObligationE19process_obligationsNtB1s_16FulfillProcesso
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x36665
rustc_driver-ebda0089c9512452.dll!RNvXNvMNtCsjEcwfPPTRNV_16rustc_hir_typeck7closureNtNtB7_7fn_ctxt6FnCtxt40deduce_closure_signature_from_predicatesNtB2_10MentionsTyINtNtCs1vmCzCB8wM7_13rustc_type_ir5visit11TypeVisitorNtNtNtCsacCxZy8tNDM_12rustc_middle2ty7
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x12cb1
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x27ec
rustc_driver-ebda0089c9512452.dll!RNvNtCshEjB07rFK3d_16rustc_query_impl8plumbing25query_key_hash_verify_all+0x155db
rustc_driver-ebda0089c9512452.dll!RINvNtNtCs8Ehw5TsioVt_18rustc_query_system5query8plumbing17try_execute_queryINtCshEjB07rFK3d_16rustc_query_impl13DynamicConfigINtNtCsfk1N0zTQ1Ep_21rustc_data_structures9vec_cache8VecCacheNtNtCs8yPNJtyUAIM_10rustc_span6def
rustc_driver-ebda0089c9512452.dll!RNvNtCshEjB07rFK3d_16rustc_query_impl8plumbing25query_key_hash_verify_all+0x1de43
rustc_driver-ebda0089c9512452.dll!RINvMs6_NtCs9VVGc76UbHk_9hashbrown3rawINtB6_8RawTableTRjuEE14reserve_rehashNCINvNtB8_3map11make_hasherBQ_uNtCszUfY7mhKel_10rustc_hash13FxBuildHasherE0ECs2JjSXZNO9KT_18rustc_hir_analysis+0x14f24
rustc_driver-ebda0089c9512452.dll!RNvCs2JjSXZNO9KT_18rustc_hir_analysis11check_crate+0xfc1
rustc_driver-ebda0089c9512452.dll!RNvNtCsbLMVCc8CHyP_15rustc_interface6passes25resolver_for_lowering_raw+0x65b1
rustc_driver-ebda0089c9512452.dll!RNvNtCsbLMVCc8CHyP_15rustc_interface6passes8analysis+0x4e
rustc_driver-ebda0089c9512452.dll!RNvMsj_NtCskXJAKvAmGc7_5alloc4syncINtB5_3ArcNtNtCsdQacyuVW3xU_13rustc_session6cstore11CrateSourceE9drop_slowCshEjB07rFK3d_16rustc_query_impl+0x6cf7a
rustc_driver-ebda0089c9512452.dll!RINvNtNtCs8Ehw5TsioVt_18rustc_query_system5query8plumbing17try_execute_queryINtCshEjB07rFK3d_16rustc_query_impl13DynamicConfigINtNtB4_6caches11SingleCacheINtNtNtCsacCxZy8tNDM_12rustc_middle5query5erase6ErasedAhj0_EEKb0_KB
rustc_driver-ebda0089c9512452.dll!RNvCshEjB07rFK3d_16rustc_query_impl12query_system+0x5377
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCNCINvMNtB6_6threadNtB1h_7Builder16spawn_unchecked_NCINvCs35sVv18JLtM_5ctrlc17set_handler_innerNCNvCs13KITJ2Zc2G_17rustc_driver_impl21install_ctrlc
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCINvNtCsbLMVCc8CHyP_15rustc_interface4util26run_in_thread_with_globalsNCINvB1e_31run_in_thread_pool_with_globalsNCINvNtB1g_9interface12run_compiler
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCNCINvMNtB6_6threadNtB1h_7Builder16spawn_unchecked_NCINvCs35sVv18JLtM_5ctrlc17set_handler_innerNCNvCs13KITJ2Zc2G_17rustc_driver_impl21install_ctrlc
rustc_driver-ebda0089c9512452.dll!ZN3std3sys3pal7windows6thread6Thread3new17h0a1b245cf1c341adE+0x1dd
KERNEL32.DLL!BaseThreadInitThunk+0x14
ntdll.dll!RtlUserThreadStart+0x21
```

</details>

<details>
<summary> backtrace 3</summary>

```
0x0000000000000000
rustc_driver-ebda0089c9512452.dll!RINvMs6_NtCs9VVGc76UbHk_9hashbrown3rawINtB6_8RawTableTINtNtNtCsacCxZy8tNDM_12rustc_middle2ty7context13InternedInSetINtNtCs1vmCzCB8wM7_13rustc_type_ir7ty_info18WithCachedTypeInfoINtNtB1U_6binder6BinderNtBT_6TyCtxtINtNtB1U_
rustc_driver-ebda0089c9512452.dll!RNvMs4_NtNtCsacCxZy8tNDM_12rustc_middle2ty7contextNtB5_13CtxtInterners16intern_predicate+0x162c
rustc_driver-ebda0089c9512452.dll!RNvXs_NtNtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits5query9normalizeNtB4_15QueryNormalizerINtNtCs1vmCzCB8wM7_13rustc_type_ir4fold18FallibleTypeFolderNtNtNtCsacCxZy8tNDM_12rustc_middle2ty7context6TyCtxtE11try_fold_ty+0
rustc_driver-ebda0089c9512452.dll!RNvXs_NtNtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits5query9normalizeNtB4_15QueryNormalizerINtNtCs1vmCzCB8wM7_13rustc_type_ir4fold18FallibleTypeFolderNtNtNtCsacCxZy8tNDM_12rustc_middle2ty7context6TyCtxtE11try_fold_ty+0
rustc_driver-ebda0089c9512452.dll!RNvXs1_NtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits7fulfillNtB5_16FulfillProcessorNtNtCsfk1N0zTQ1Ep_21rustc_data_structures17obligation_forest19ObligationProcessor18process_obligation+0x1e29
rustc_driver-ebda0089c9512452.dll!RINvMs0_NtCsfk1N0zTQ1Ep_21rustc_data_structures17obligation_forestINtB6_16ObligationForestNtNtNtCs9P1nyiK2aZF_21rustc_trait_selection6traits7fulfill26PendingPredicateObligationE19process_obligationsNtB1s_16FulfillProcesso
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x36665
rustc_driver-ebda0089c9512452.dll!RNvXNvMNtCsjEcwfPPTRNV_16rustc_hir_typeck7closureNtNtB7_7fn_ctxt6FnCtxt40deduce_closure_signature_from_predicatesNtB2_10MentionsTyINtNtCs1vmCzCB8wM7_13rustc_type_ir5visit11TypeVisitorNtNtNtCsacCxZy8tNDM_12rustc_middle2ty7
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x12cb1
rustc_driver-ebda0089c9512452.dll!RNvCsjEcwfPPTRNV_16rustc_hir_typeck6typeck+0x27ec
rustc_driver-ebda0089c9512452.dll!RNvNtCshEjB07rFK3d_16rustc_query_impl8plumbing25query_key_hash_verify_all+0x155db
rustc_driver-ebda0089c9512452.dll!RINvNtNtCs8Ehw5TsioVt_18rustc_query_system5query8plumbing17try_execute_queryINtCshEjB07rFK3d_16rustc_query_impl13DynamicConfigINtNtCsfk1N0zTQ1Ep_21rustc_data_structures9vec_cache8VecCacheNtNtCs8yPNJtyUAIM_10rustc_span6def
rustc_driver-ebda0089c9512452.dll!RNvNtCshEjB07rFK3d_16rustc_query_impl8plumbing25query_key_hash_verify_all+0x1de43
rustc_driver-ebda0089c9512452.dll!RINvMs6_NtCs9VVGc76UbHk_9hashbrown3rawINtB6_8RawTableTRjuEE14reserve_rehashNCINvNtB8_3map11make_hasherBQ_uNtCszUfY7mhKel_10rustc_hash13FxBuildHasherE0ECs2JjSXZNO9KT_18rustc_hir_analysis+0x14f24
rustc_driver-ebda0089c9512452.dll!RNvCs2JjSXZNO9KT_18rustc_hir_analysis11check_crate+0xfc1
rustc_driver-ebda0089c9512452.dll!RNvNtCsbLMVCc8CHyP_15rustc_interface6passes25resolver_for_lowering_raw+0x65b1
rustc_driver-ebda0089c9512452.dll!RNvNtCsbLMVCc8CHyP_15rustc_interface6passes8analysis+0x4e
rustc_driver-ebda0089c9512452.dll!RNvMsj_NtCskXJAKvAmGc7_5alloc4syncINtB5_3ArcNtNtCsdQacyuVW3xU_13rustc_session6cstore11CrateSourceE9drop_slowCshEjB07rFK3d_16rustc_query_impl+0x6cf7a
rustc_driver-ebda0089c9512452.dll!RINvNtNtCs8Ehw5TsioVt_18rustc_query_system5query8plumbing17try_execute_queryINtCshEjB07rFK3d_16rustc_query_impl13DynamicConfigINtNtB4_6caches11SingleCacheINtNtNtCsacCxZy8tNDM_12rustc_middle5query5erase6ErasedAhj0_EEKb0_KB
rustc_driver-ebda0089c9512452.dll!RNvCshEjB07rFK3d_16rustc_query_impl12query_system+0x5377
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCNCINvMNtB6_6threadNtB1h_7Builder16spawn_unchecked_NCINvCs35sVv18JLtM_5ctrlc17set_handler_innerNCNvCs13KITJ2Zc2G_17rustc_driver_impl21install_ctrlc
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCINvNtCsbLMVCc8CHyP_15rustc_interface4util26run_in_thread_with_globalsNCINvB1e_31run_in_thread_pool_with_globalsNCINvNtB1g_9interface12run_compiler
rustc_driver-ebda0089c9512452.dll!RINvNtNtCsaor5jccb5ur_3std3sys9backtrace28___rust_begin_short_backtraceNCNCNCINvMNtB6_6threadNtB1h_7Builder16spawn_unchecked_NCINvCs35sVv18JLtM_5ctrlc17set_handler_innerNCNvCs13KITJ2Zc2G_17rustc_driver_impl21install_ctrlc
rustc_driver-ebda0089c9512452.dll!ZN3std3sys3pal7windows6thread6Thread3new17h0a1b245cf1c341adE+0x1dd
KERNEL32.DLL!BaseThreadInitThunk+0x14
ntdll.dll!RtlUserThreadStart+0x21
```

</details>

Might be worth noting that the compiler memory keeps steadily and rather rapidly increasing.

@rustbot label: +I-hang +T-compiler +T-types

**Comment by moxian** [comment]
Correction: The bisection above is only valid for the original code (`impl<T, U> Trait for W<(<&Self as IntoIterator>, W<U>)>`), but not for minimization (`impl<T, U> Trait for W<()>`).

The latter bisects to nightly-2024-03-08 (`1.78.0-nightly`) instead


<details>
<summary>cargo-bisect-rustc output</summary>

```
********************************************************************************
Regression in nightly-2024-03-08
********************************************************************************

converted 2024-03-07 to 7d3702e472b99be0f5de6608dd87af1df8f99428
converted 2024-03-08 to 9c3ad802d9b9633d60d3a74668eb1be819212d34
get_commits_between returning commits, len: 12
  commit[0] 2024-03-06: Auto merge of #122111 - matthiaskrgr:rollup-qq4v6gs, r=matthiaskrgr
  commit[1] 2024-03-07: Auto merge of #122117 - matthiaskrgr:rollup-3yrv3j6, r=matthiaskrgr
  commit[2] 2024-03-07: Auto merge of #122113 - matthiaskrgr:rollup-5d1jnwi, r=matthiaskrgr
  commit[3] 2024-03-07: Auto merge of #122054 - weihanglo:update-cargo, r=weihanglo
  commit[4] 2024-03-07: Auto merge of #121866 - Kobzol:opt-dist-find-llvm, r=Mark-Simulacrum
  commit[5] 2024-03-07: Auto merge of #121154 - oli-obk:track_errors11, r=estebank
  commit[6] 2024-03-07: Auto merge of #121985 - RalfJung:interpret-return-place, r=oli-obk
  commit[7] 2024-03-07: Auto merge of #12429 - Alexendoo:redundant-field-names-macro-ctxt, r=Manishearth
  commit[8] 2024-03-07: Auto merge of #122139 - GuillaumeGomez:rollup-37vtwsc, r=GuillaumeGomez
  commit[9] 2024-03-07: Auto merge of #12431 - flip1995:rustup, r=flip1995
  commit[10] 2024-03-07: Auto merge of #122149 - flip1995:clippy-subtree-update, r=Manishearth
  commit[11] 2024-03-07: Auto merge of #119199 - dpaoliello:arm64ec, r=petrochenkov
ERROR: no CI builds available between 7d3702e472b99be0f5de6608dd87af1df8f99428 and 9c3ad802d9b9633d60d3a74668eb1be819212d34 within last 167 days
```

</details>

**Comment by apiraino** [comment]
I bisected the first reproducible again and I got:

> Regression in https://github.com/rust-lang-ci/rust/commit/a7a83cca2639eda504d971561a7409087536f816
> The PR introducing the regression in this rollup is #132090: Stop being so bail-y in candidate assembly

I am not sure I fully grasp the opening comment for #132090 but I *think* it's related to the hanging, maybe cc @compiler-errors ?

@rustbot label -I-prioritize +P-medium
